tests/test_shell.py

```diff
@@ -36,20 +36,21 @@ class ExitCodeError(RuntimeError):
     pass
 
 
-class BasicShellTest(unittest.TestCase):
+class TestBasicShell():
 
-    def setUp(self):
+    @pytest.fixture(autouse=True)
+    def setUp(self, request):
         pythonpath = os.environ.get('PYTHONPATH')
         if pythonpath is None:
             pythonpath = [os.getcwd()]
         else:
             pythonpath = [os.getcwd()] + pythonpath.split(':')
         os.environ['PYTHONPATH'] = ':'.join(pythonpath)
         self.tmpdir = TemporaryDirectory(prefix='signac_')
-        self.addCleanup(self.tmpdir.cleanup)
         self.cwd = os.getcwd()
-        self.addCleanup(self.return_to_cwd)
         os.chdir(self.tmpdir.name)
+        request.addfinalizer(self.tmpdir.cleanup)
+        request.addfinalizer(self.return_to_cwd)
```

Comment:
I believe that the finalizer should be added immediately after the corresponding call, just in case any of the following calls fails.